src/util/system.cpp

```diff
@@ -319,6 +318,27 @@ ArgsManager::ArgsManager() :
       "-port", "-bind",
       "-rpcport", "-rpcbind",
       "-wallet",
+    },
+
+    m_allowed_negated_args{
```

Comment:
The problem described in the PR is real but I don't think the change implemented is the simplest or best fix. Having a new whitelist of options located in a separate part of the code from where the options are used and defined adds complexity, inconsistency, and [action at distance](https://en.wikipedia.org/wiki/Action_at_a_distance_(computer_programming)) to already very confusing code.

If someone specifies `-nopid`, it is wrong to create a pid file called `0`, but a simpler and more useful fix would be to be not write a pid file instead of adding a new logic in a different part of the code to print a "negating not allowed" error.

But the `-pid` argument case actually seems atypical here. More cases where negated arguments are handled incorrectly (like `-incrementalrelayfee`) seem like simple misuses of the IsArgSet function, which is misnamed and doesn't do what people think it does. I'd replace the `IsArgSet` function with separate `ArgHasValue` and  `ArgIsSpecified` functions and use `ArgHasValue` instead of `ArgIsSpecified` everywhere or almost everywhere the code currently using ArgIsSet.

Reply:
`m_allowed_negated_args` is used in the same manner as `m_network_only_args` (see #11862) and it seems consistent, IMO.

Reply:
> adds complexity, inconsistency, and action at distance to already very confusing code

Agree with @ryanofsky. I'd rather see an approach like https://github.com/bitcoin/bitcoin/pull/16416#issuecomment-514120375

Beside the fact that `m_network_only_args` is very small, I think it could be improved like my comment above - move that flag to the `Arg` structure.

Currently `ArgsManager::Arg` has:
```cpp
        bool m_debug_only;
```
And I think it could be changed to something like:
```cpp
   enum Flags {
      NONE = 0x00,
      DEBUG_ONLY = 0x01,
      NETWORK_ONLY = 0x2,
      ALLOW_NEGATED = 0x04,      
   };

   Flags m_flags;
```